### ISSUE TITLE
Date localization

### ISSUE THREAD

**Issue by JaneX8** [navidrome/navidrome#229]
Can this and all date notations have localization support. I suggest to use Moment.js it probably integrates perfectly with their localization support with React. Based on chose language or separate notation setting.

![image](https://user-images.githubusercontent.com/5116641/80376899-c7675300-889a-11ea-9c84-50225b850a70.png)

**Comment by deluan** [comment]
Will check the options. I don't want to add moment.js if I can avoid, as it is a huge library and will add a lot of fat to our js bundle

**Comment by ElleshaHackett** [comment]
Understandable. Maybe the used ra- translation libs have such options?

**Comment by tcurdt** [comment]
Or just use an ISO 8601? Be part of the positive change in this world ;)

**Comment by deluan** [comment]
I think ISO8601 could be an intermediate step, better than what we have now, but I don't like it as it is less readable than the local convention, IMO. 

Ex Localtime: 9:37 EST:
```
ISO:    2020-05-16T13:37:24Z
en_CA:  16/05/2020 09:37:24
```

**Comment by tcurdt** [comment]
It was a bit tongue in cheek as that depends on the local convention :-) If you'd remove the "T" and "Z" I think at least most people across the pond (or even world) would be fine.

https://en.wikipedia.org/wiki/Date_format_by_country

Maybe just offer it as a setting? Seems like the most lightweight and flexible option.
I for one often don't use the standard format often given for my locale.

**Comment by aldrinjenson** [comment]
We use React-admin dateField component to render these dates. Instead of using any extra libraries, we could instead pass in some dateOptions as props to customize the result based on the locale to a certain good extent.
It supports same options as those passed as second argument to the JS [Date.toLocaleString](https://developer.mozilla.org/fr/docs/Web/JavaScript/Reference/Global_Objects/Date/toLocaleDateString)

Here is an example where each field is given locales of 'en', 'es' and 'de' respectiviely.
![image](https://user-images.githubusercontent.com/53407417/113581566-ec387e80-9644-11eb-971e-5c5aebe6f546.png)

**Comment by deluan** [comment]
Yeah, let's try passing the current language selected. It should work as we are using the ISO-639-1 (two letters language code) for the language. You can retrieve the current selected language using `useLocale()`

**Comment by tcurdt** [comment]
Maybe I am too exotic to count as a normal user - but my language always is `en` and I still want a 24h clock.

**Comment by tcurdt** [comment]
> Here is an example where each field is given locales of 'en', 'es' and 'de' respectiviely.
> ![image](https://user-images.githubusercontent.com/53407417/113581566-ec387e80-9644-11eb-971e-5c5aebe6f546.png)

That dateField is wrong though. The German locale should use 24h.

**Comment by aldrinjenson** [comment]
> That dateField is wrong though. The German locale should use 24h - and I pretty sure the same is true for Spain.

Ahh, this can be easily fixed. For the example image, I had passed the 12hour option as true. Didn't know before reading the Wiki article(thanks for this!) that 24h clock is quite common in a lot of countries.

> Maybe I am too exotic to count as a normal user - but my language always is en and I still want a 24h clock.

Hmm, we could add another dropdown for clock type in the personal page with three options -> Default(as calculated from the locale), 12hr, 24hr.
I'm not sure of the translation part though. Since it'd be small statements ('Clock Type' , 'Default'), can Google translate be reliably used?

**Comment by deluan** [comment]
You should not override the `hour12` option, so then it uses the default for that locale. 

The idea of adding a config option is nice. A "Datetime format" with two options:
- Default (not touching the `hour12` option)
- ISO 8601 (modified as per @tcurdt [suggestions](https://github.com/navidrome/navidrome/issues/229#issuecomment-629661551)

For translation you will need to use the `useTranslate()` hook, and add the strings to `en.json`. Check the other settings for examples.

**Comment by anarcat** [comment]
> I think ISO8601 could be an intermediate step, better than what we have now

god please start with that.

> but I don't like it as it is less readable than the local convention, IMO.

what do you mean by "local convention" here?

somehow i absolutely refuse to accept that MM/DD/YYY is more readable than anything. where I come from, it's absolutely ambiguous and it's impossible to tell what that date is. it drives me absolutely crazy every time i see the remnant of this. i thought we had gotten rid of this decades ago.

ISO8601 is 33 years old this year. maybe it's time we make it the default? if you really, really want to use some obscure dating system your country insists on adopting regardless of standardization, that's fine, but could we keep things readable by default? :)

> https://en.wikipedia.org/wiki/Date_format_by_country

incredibly, the first row in that table includes a lot of europe, most of africa and all of south america, for a whopping 2+ billion people which use not YYYY-MM-DD, but .... DD-MM-YYYY, which I also find absurd. but i guess it depends where you come from too.

anyways, i think the point of that page was to show that only a tiny fraction of people actually use M/D/Y as a convention (ie. the US, basically) and it's unfair to set that as a default.

sorry for jumping into that old issue, but i figured it would be nice to just fix this in what i feel is the easiest way at first. And you don't have to follow ISO8601 litterally and add that `T` separator. You could just say the date follows ISO8601 and the time follows. Or, if you really want to be pedantic about conformity, you could just say you follow [rfc3339](https://datatracker.ietf.org/doc/html/rfc3339) instead which *does* allow space as a separator...

**Comment by tcurdt** [comment]
This has been open for a while.

I assume it should be added here?

https://github.com/navidrome/navidrome/blob/master/ui/src/i18n/en.json#L105

As for existing formatting implementations:
- luxon too big
- moment.js 18kb
- date-fns should not add much overhead

"With the function-per-file style, you can pick just what you need and stop bloating your project with useless functionality.
It works well with modern module bundlers such as webpack, Browserify, and Rollup and also supports tree-shaking."

But there is also a simple date formatting impl:
https://stackoverflow.com/a/52789490/33165

**Comment by SteveDinn** [comment]
Came here to log a similar issue, but for the "Last Played" field in the Album view. I would love to be able to view things in 24-hour time.

![image](https://github.com/navidrome/navidrome/assets/3911507/4a4e8cde-27e2-45a8-952c-10f78b80e5c3)

**Comment by JaneX8** [comment]
Time localization (12-hour vs. 24-hour notation) is a must too. I agree.

**Comment by tcurdt** [comment]
Please un-stale.

**Comment by tcurdt** [comment]
Please un-stale.

**Comment by deluan** [comment]
This will be handled in the new UI

**Comment by tcurdt** [comment]
unstale - still relevant until the new UI hits

**Comment by tcurdt** [comment]
unstale 🙄

**Comment by tcurdt** [comment]
unstale

**Comment by brianpierson2020** [comment]
6 years 🥲

**Comment by tcurdt** [comment]
> 6 years 🥲

😔 I have the feeling the built-in frontend is just dead. But 6y is pretty crazy.

**Comment by deluan** [comment]
Not dead, I use it everyday :)
But yeah, the new UI is taking longer than I expected, so here's the fix: #6160 

Please test it and let me know.
